File: implementation/codeJava/src/main/java/Facture.java
import java.util.Date;

public class Facture {
   private int numFacture;
   private Date dateFacture;
   private Float montantFacture;
   private String etatFacture;
   
   public Commande commande;
   public Caissier caissier;
   
   public Caissier getCaissier() {
      return caissier;
   }

   public void setCaissier(Caissier newCaissier) {
      if (this.caissier == null || !this.caissier.equals(newCaissier))
      {
         if (this.caissier != null)
         {
            Caissier oldCaissier = this.caissier;
            this.caissier = null;
            oldCaissier.removeFacture(this);
         }
         if (newCaissier != null)
         {
            this.caissier = newCaissier;
            this.caissier.addFacture(this);
         }
      }
   }

}

File: restorino/lib/metier/models/facture.dart
import 'dart:ffi';

import 'caissier.dart';
import 'commande.dart';

class Facture {
   int numFacture;
   DateTime dateFacture;
   Float montantFacture;
   String etatFacture;

   Commande commande;
   Caissier caissier;


   Facture(
       this.numFacture,
       this.dateFacture,
       this.montantFacture,
       this.etatFacture,
       this.commande,
       this.caissier);


}